### PR TITLE
feat: Phoenix.Stream for Table component

### PR DESCRIPTION
**Pull request by phgrey**

### CODE REVIEW

**Review by tkullisaar — APPROVED** [review]

File: lib/moon/design/table.ex [hunk 1]
```diff
@@ -182,4 +185,13 @@ defmodule Moon.Design.Table do
...
   end
+
+  defp stream_data(%{items: stream = %Phoenix.LiveView.LiveStream{}}) do
+    stream
+  end
+
+  defp stream_data(%{items: items, sort: sort}) when is_list(items) do
+    Enum.with_index(items |> add_index_as |> sort_items(sort))
+    |> Enum.map(fn {item, key} -> {key, item} end)
+  end
```
Comment (samaraoliveiram) [inline lib/moon/design/table.ex:196]: Hey just a suggestion, why not write like this to be more readable?
```suggestion
  defp stream_data(%{items: items, sort: sort}) when is_list(items) do
   items 
   |> add_index_as 
   |> sort_items(sort)
   |> Enum.with_index(fn item, key -> {key, item} end)
  end
```
On `with_index` you can pass a function that already return the way you want the tuple, doing like that you save 1 iteration over the list as well.

**Comment by samaraoliveiram** [comment]
Looking good, thanks!

File: lib/moon_web/examples/design/table_example/stream.ex [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          name: "Name \#{x}",
+          created_at: DateTime.add(DateTime.utc_now(), -3600 + x)
+        }
+      end)
+    end
+
+    def render(assigns) do
+      ~F\"""
+      <div>
+        <Table items={model <- @streams.models} selected={nil} body_attrs={%{"phx-update" => "stream"}}>
```
Comment (samaraoliveiram) [inline lib/moon_web/examples/design/table_example/stream.ex:68]: hey just a question:
If I pass to `body_attrs` this : 
`body_attrs={%{"phx-viewport-bottom" => "{@page_info["hasNextPage"] && "next-page"}}"` 
the final result is going to be `phx-viewport-bottom={@page_info["hasNextPage"] && "next-page"}}` ?

Comment (phgrey) [reply lib/moon_web/examples/design/table_example/stream.ex:68]: i believe it will be a syntax error))))))

**Pull request merged**